src/Main.java

```diff
@@ -1,278 +1 @@
-package com.company;
 
-import javax.swing.text.StyledEditorKit;
-import java.util.Scanner;
-import java.io.File;
-import java.io.*;
-import java.util.Queue;
-import java.util.*;
-
-class Cell{
-    int x;
-    int y;
-    int gCost;
-    int hCost;
-    Cell parent;
-
-    public Cell(int x, int y, Cell parent){
-        this.x = x;
-        this.y = y;
-        this.parent = parent;
-    }
-
-    public Cell(int x, int y, Cell parent, int gCost, int hCost){
-        this.x = x;
-        this.y = y;
-        this.parent = parent;
-        this.gCost = gCost;
-        this.hCost = hCost;
-    }
-
-    public Cell getParent(){
-        return this.parent;
-    }
-}
-
-class MazeSolver {
-
-    //4 possible moves
-    static int[] move_x = {1, -1, 0, 0};
-    static int[] move_y = {0, 0, 1, -1};
-
-    // Using DFS algorithm, does not guarantee shortest path
-    static Boolean MazeBackTracker(char[][] maze, Cell cell, int m, int n){
-        // Boundary conditions
-        if (!isValidMove(maze, cell, m, n)) return false;
-
-        // Found exit
-        if (maze[cell.x][cell.y] == 'E') return true;
-
-        // Mark current node as visited
-        maze[cell.x][cell.y] = 'X';
-
-        //Recursively visit the next nodes
-        for (int move = 0; move < 4; move++){
-            Cell nextCell = new Cell(cell.x + move_x[move], cell.y + move_y[move], cell);
-            if (MazeBackTracker(maze, nextCell, m, n)) return true;
-        }
-
-        // Backtrack
-        maze[cell.x][cell.y] = ' ';
-        return false;
-    }
-
-    // Using BFS algorithm
-    static Cell BFS(char[][] maze, Cell cell, int m, int n){
```

Comment:
In this case, BFS should be 'bFS' in the naming convention.

Reply:
Also - is there a reason to pass around 'm' and 'n' when the size of the maze will give us those values?